controllers/templatesync/template_sync.go

```
@@ -1740,3 +1754,41 @@ func getDupName(pol *policiesv1.Policy) string {
 
 	return ""
 }
+
+// In Gatekeeper 3.17 and later, if a ConstraintTemplate contains a Rego syntax error,
+// it is still created, but its status field will have "created: false" instead of failing outright.
+// This function retrieves the "status.created" field by recursively checking the status
+// until the field is present.
+func getConstraintTemplateStatus(ctx context.Context, name string, res dynamic.ResourceInterface) (bool, error) {
+	template, err := res.Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	created, ok, err := unstructured.NestedBool(template.Object, "status", "created")
+	if !ok || err != nil {
+		time.Sleep(500 * time.Millisecond)
+
+		return getConstraintTemplateStatus(ctx, name, res)
+	}
```

If gatekeeper isn't running, then this will recursively retry forever (or until it crashes), waiting for the status to be populated. That could prevent any other work from getting done.

Maybe it should return an error, which gets surfaced as a template-error in the policy, and cause the Reconcile to be re-queued to wait for the status to be populated.

---

If gatekeeper is not running, ConstraintTemplate will be failed in `rec.get`

---

I fixed